client/src/schema/repository.ts

```diff
@@ -321,6 +321,15 @@ export class BaseClient<D extends Record<string, Repository<any>>> {
 
     this.options = options;
     this.#links = links;
+
+    const factory = options.repositoryFactory || new RestRespositoryFactory();
+
+    this.db = new Proxy({} as D, {
+      get: (_target, prop) => {
+        if (typeof prop !== 'string') throw new Error('Invalid table name');
+        return factory.createRepository(this, prop);
```

Comment:
I wasn't 100% sure about this ``this``, but since it's an arrow function that does not have ``this``, the parent one is persisted (still the base client).